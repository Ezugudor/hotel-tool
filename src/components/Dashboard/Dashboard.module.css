.Header{
    background:#fff;
    padding:30px;
    font-size:30px
}

.StatsCont{
    display:flex;
    justify-content:space-between;
    padding:25px
}

.PlaceholderWrapper{
   display:flex;
   justify-content:space-around;
   padding-top:50px;
   /* height:80% */
}
.PlaceholderCont{
    align-self:center;
    width:70%;
    height:40vh;
    display:flex;
    flex-direction:column;
    align-items:center
}
.PlaceholderIconCont{
    padding:20px;
}

.PlaceholderIconCont i{
   font-size:7vw;
   color:#777
}

.PlaceholderCont h4{
    font-size:3vw;
    color:#999
 }
 .PlaceholderInfo{
    font-size:2vw;
    color:#aaa;
    text-align:center
 }
 .TableCont{
     /* background:red; */
     /* height:100vh */
     padding:0 25px
 }
 .TableHeader{
     display:flex;
     justify-content:space-between;
     background:#fff;
     padding:30px;
     margin-bottom:5px;
     align-items:center
 }
 .TableHeader input{
    background:#f1f1f1;
    border:1px solid #ccc;
    padding:10px;
    border-radius:4px
}
 .TableCont table{
     width:100%;

 }

 .TableCont table thead th, .TableCont table tbody td{
   background:#fff;
   padding:10px
    
}
.Todo{
    color:#aaa
}