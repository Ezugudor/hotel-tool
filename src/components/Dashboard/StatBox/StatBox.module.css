.Container{
    background:#fff;
    padding:10px;
    width:30%;
    display:flex;
    align-items:center;
    border-radius:4px
}
.IconCont{
    padding:1.5vw
}


.IconCont i{
    font-size:4vw
}


.StatsCont{
    display:flex;
    justify-content:space-between;
}
.InfoCont h4{
    margin:0;
    font-size:1.5vw;
    margin-bottom:1rem;
    font-weight:normal;
   
}

.Info{
    display:flex;
    justify-content:space-between;
    font-size:1.3vw;
    margin-bottom:5px;
}


.Label{
    margin-right:30px;
    color:#777
}
.Value{
    font-weight:bold
}