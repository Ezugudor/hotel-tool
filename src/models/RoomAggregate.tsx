export interface RoomAggregate {
    roomCount: number;
    amount: number;
}