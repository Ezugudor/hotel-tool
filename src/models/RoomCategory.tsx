export enum RoomCategory {
    PREMIUM = "premium",
    ECONOMY = "economy",
    NON_PREMIUM = "non_premium", //economy + others not meeting up to premium benchmark;
}