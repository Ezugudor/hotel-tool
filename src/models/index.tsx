export * from "./RoomAggregate"
export * from "./RoomCategory"