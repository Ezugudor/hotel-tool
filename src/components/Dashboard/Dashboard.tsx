import React, { useContext, useEffect } from "react";
import { AvailableRoomsContext, CustomerContext } from "../../context";
import { useAvailableRoomProfitEstimator } from "../../hooks/useAvailableRoomProfitEstimator";
import { StatBox } from "./StatBox";


export const Dashboard: React.FunctionComponent = () => {

    const [estimatedResult, setEstimatedResult] = useAvailableRoomProfitEstimator();
    const customerBids = useContext(CustomerContext);
    const [availableRooms] = useContext(AvailableRoomsContext);

    useEffect(() => {
        const CustomerBidsCopy = [...customerBids];
        setEstimatedResult(CustomerBidsCopy, parseInt(`${availableRooms.economy}`), parseInt(`${availableRooms.premium}`));
    }, [availableRooms, customerBids])


    return (
        <div>
            <div id="stat-box-cont">
                <StatBox data={estimatedResult.total} />
                <StatBox data={estimatedResult.economy} />
                <StatBox data={estimatedResult.premium} />
            </div>
        </div>
    )
}
