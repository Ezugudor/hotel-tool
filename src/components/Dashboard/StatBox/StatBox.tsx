import React from "react";
import { RoomAggregate } from "../../../models";
import styles from "./StatBox.module.css"

export interface StatBoxProps {
    title?: string;
    icon?: string;
    color?: string;
    data: RoomAggregate
}
export const StatBox: React.FC<StatBoxProps> = (props) => {
    const { data } = props;
    return (
        <div className={styles.Container}>
            <div className={styles.IconCont}><i className={`ion ${props?.icon}`} style={{ color: props?.color }}></i></div>
            <div className={styles.InfoCont}>
                <h4>{props?.title}</h4>
                <div className={styles.Info}>
                    <span className={styles.Label}>Rooms:</span>
                    <span className={`${styles.Value} room`}>{data.roomCount}</span>
                </div>
                <div className={styles.Info}>
                    <span className={styles.Label}>Amount:</span>
                    <span className={styles.Value}>â‚¬ <span className="amount">{data.amount.toLocaleString()}</span></span>
                </div>
            </div>
        </div>
    )
}
