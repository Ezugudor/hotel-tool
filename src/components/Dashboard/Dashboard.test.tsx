import React from 'react';
import { shallow, ShallowWrapper } from 'enzyme';
import { Dashboard } from './Dashboard';
import { StatBox, StatBoxProps } from './StatBox';
import { RoomAggregate } from '../../models';

let DashboardCont: ShallowWrapper;
let StatBoxCont: ShallowWrapper;
let StatBoxWrapper: ShallowWrapper<StatBoxProps>;

beforeAll(() => {
    DashboardCont = shallow(<Dashboard />);
    StatBoxCont = DashboardCont.find('#stat-box-cont').hostNodes();
    StatBoxWrapper = StatBoxCont.find(StatBox);
});


describe("Dashboard Component Test", () => {
    it('should match with existing snapshot', () => {
        expect(DashboardCont.html()).toMatchSnapshot();
    })

    it("should render 3 StatBox components inside an element with id 'stat-box-cont' ", () => {
        expect(StatBoxCont).toHaveLength(1)
        expect(StatBoxWrapper).toHaveLength(3)
    })

    it("Each of the 3 StatBox components should have a prop 'data' of type 'Aggregate'", () => {
        const statBox1: RoomAggregate = StatBoxWrapper.at(0).prop('data');
        const statBox2: RoomAggregate = StatBoxWrapper.at(1).prop('data');
        const statBox3: RoomAggregate = StatBoxWrapper.at(2).prop('data');
        const expected = ["roomCount", "amount"];
        expect(Object.keys(statBox1)).toEqual(expect.arrayContaining(expected))
        expect(Object.keys(statBox2)).toEqual(expect.arrayContaining(expected))
        expect(Object.keys(statBox3)).toEqual(expect.arrayContaining(expected))
    })

})


