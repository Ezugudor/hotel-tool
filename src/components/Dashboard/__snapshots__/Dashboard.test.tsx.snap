// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dashboard Component Test should match with existing snapshot 1`] = `"<div><div id=\\"stat-box-cont\\"><div><div><span>Rooms:</span><span>0</span></div><div><span>Amount:</span><span>0</span></div></div><div><div><span>Rooms:</span><span>0</span></div><div><span>Amount:</span><span>0</span></div></div><div><div><span>Rooms:</span><span>0</span></div><div><span>Amount:</span><span>0</span></div></div></div></div>"`;
